packages/jest-runtime/src/transform.js

```diff
@@ -276,7 +246,12 @@ const cachedTransformAndWrap = (
   if (preprocessor && shouldPreprocess(filename, config)) {
     result = preprocessor.process(result, filename, config);
   }
-  if (shouldInstrument(filename, config)) {
+
+  // That means that the preprocessor has a custom instrumentation
+  // logic and will handle it based on `config.collectCoverage` option
+  const preprocessorWillInstrument = preprocessor && preprocessor.INSTRUMENTS;
+
+  if (!preprocessorWillInstrument && shouldBeCovered(filename, config)) {
```

Comment:
we are calling shouldBeCovered three times (+ one more in the preprocessor) with this change. Can we cache the result and only call it once?